### PR TITLE
Vertical filter UX follow ups

### PR DESCRIPTION
**Pull request by sofiamatulis**
**PR Summary:** 

- [x] * Mobile drawer, cannot scroll
- [x] * Vertical filter: Temp fix, Update buttons to look like secondary style and align everything -  Tertiary button style button outline is lost, not visible, button don’t look like a button (No JS) - Update to secondary
- [x] * [#1558 comment](https://github.com/Shopify/dawn/issues/1558#issuecomment-1083992298) - Drawer: Filter pills [wrap too soon](https://screenshot.click/22-03-c1v59-b5whh.png) compared to [horizontal layout](https://screenshot.click/22-03-twda6-fquyp.png)
- [x] * [#1558 comment](https://github.com/Shopify/dawn/issues/1558#issuecomment-1083992298) - I understand we are scaling the pills [from this comment](https://github.com/Shopify/dawn/pull/1443#issuecomment-1081883513), but the Remove all gets misaligned when we do it, we can adjust it still.
- [x]       Search template * Mobile Drawer → Filter button and drawer visible when filters are not enabled and sort is enabled.
- [x]    Collection template     *Mobile drawer Sort not showing up when filters is disabled

**Demo links**
- [Editor](https://os2-demo.myshopify.com/admin/themes/127787794454/editor)

**Checklist**
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Notified the [help.shopify.com](https://help.shopify.com) team about updates to theme settings (current contact: Kimli)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by sofiamatulis** [comment]
Quick comment: The sort by: moves depending on horizontal and drawer https://screenshot.click/31-59-jt9kb-le6wb.mp4

**Review by melissaperreault — CHANGES_REQUESTED** [review]
Thank you Sofia! We've got still a few adjustments to make! 🙏 

**What I tested:**
- [x] Mobile drawer, cannot scroll
- [x] Vertical filter: Temp fix, Update buttons to look like secondary style and align everything
- [x] Tertiary button style button outline is lost, not visible, button don’t look like a button (No JS) - Update to secondary
- [x] Drawer: Filter pills [wrap too soon](https://screenshot.click/22-03-c1v59-b5whh.png) compared to [horizontal layout](https://screenshot.click/22-03-twda6-fquyp.png)
- [x] I understand we are scaling the pills, but the Remove all gets misaligned when we do it, we can adjust it still.
- [x] Search template * Mobile Drawer → Filter button and drawer visible when filters are not enabled and sort is enabled.
- [x] Collection template *Mobile drawer Sort not showing up when filters is disabled

##  Findings

### Vertical layout
**Both Collection and Search**
- [x] When ❌ Enable filtering but ☑️ Enable sorting; fill in the [empty space](https://screenshot.click/22-04-1afw4-0wdap.png). (I wonder if we should just have the sorting in that column, but the outcome isn't ideal now regardless). 

### No JS

#### Collection template
_Vertical layout, ☑️ Enable filtering, ☑️ Enable sorting_
- [x] [Both buttons would be secondary style](https://screenshot.click/22-04-xfgfp-wlzn2.png) for the temporary fix

_Vertical layout, ❌ Enable filtering, ☑️ Enable sorting_
- [x] Button [to be secondary style](https://screenshot.click/22-04-g3vnp-o2gu5.png)
- [ ] Sort dropdown and button [are visible on mobile now. It should not be duplicated](https://screenshot.click/22-04-ivdpm-sfl96.png)

_Horizontal layout, ❌ Enable filtering, ☑️ Enable sorting_
-  [ ] To apply [secondary button style](https://screenshot.click/22-04-idkn5-zkjnk.png)

_Horizontal layout, ☑️ Enable filtering, ☑️ Enable sorting_
-  [ ] To apply [secondary button style](https://screenshot.click/22-04-46lmm-76flt.png)

_Drawer layout, ☑️ Enable filtering, ❌ Enable sorting_ 
- [ ] The [product count alignment is misplaced](https://screenshot.click/22-04-a49q6-tjvki.png). It also have the [filter pills not place properly](https://screenshot.click/22-04-f419h-ix8hl.png).
   - [ ] This is also happening WITH JS over here: [product count alignment is misplaced](https://screenshot.click/22-04-3lwfq-3pp1z.png)  and [here](https://screenshot.click/22-04-ryt1f-xkwa3.png)

_Drawer layout, ☑️ Enable filtering, ☑️ Enable sorting_
- [ ] To [apply secondary button style](https://screenshot.click/22-04-7v31g-23qe8.png)

####  Search template

_Vertical layout, ☑️ Enable filtering, ☑️ Enable sorting_
- [ ] [Both buttons would be secondary style](https://screenshot.click/22-04-iicjt-jl8ev.png)

_Vertical layout, ❌ Enable filtering, ☑️ Enable sorting_
- [ ] Button [to be secondary style](https://screenshot.click/22-04-d8zaz-w190r.png)
- [ ] Sort dropdown and button [are visible on mobile now. It should not be duplicated](https://screenshot.click/22-04-utw3n-n650v.png)

_Horizontal layout, ☑️ Enable filtering, ☑️ Enable sorting_
-  [ ] To apply [secondary button style](https://screenshot.click/22-04-gre7d-j36ab.png)

_Horizontal layout, ❌ Enable filtering, ☑️ Enable sorting_
-  [ ] To apply [secondary button style](https://screenshot.click/22-04-v15uw-l3nk4.png)


_Drawer layout, ☑️ Enable filtering, ❌ Enable sorting_ 
- [ ] The [product count alignment is misplaced](https://screenshot.click/22-04-xsju7-x7d1w.png). It also have the [filter pills not place properly](https://screenshot.click/22-04-e7nt7-5hqld.png).
   - [ ] This is also happening WITH JS over here: [product count alignment is misplaced](https://screenshot.click/22-04-r76le-9t0ce.png)  and [here](https://screenshot.click/22-04-mv731-mut3s.png)

_Drawer layout, ☑️ Enable filtering, ☑️ Enable sorting_
- [ ] To [apply secondary button style](https://screenshot.click/22-04-bae0i-y8lfy.png)
- [ ] The filter pills [wrap too soon over here](https://screenshot.click/22-04-w9u64-t0r1i.png)

File: snippets/facets.liquid
Comment (martinamarien) [inline snippets/facets.liquid:264]: We will need to apply this to a few more buttons. I included a table in a main comment that indicates where we may need to change primary and tertiary to secondary.

**Comment by melissaperreault** [comment]
> I'll let Meli confirm this one. I'm not too sure if it's fixed or not

@martinamarien  Yes it is fixed, the smaller Removal all link is [aligned with the Filter header now](https://screenshot.click/22-04-2zhag-mgq1r.png), could receive margin-top: 0.2rem to be eagle eye but this will be tackled from another polish issue! 🙌 

All buttons look like secondary with no JS now ! 🎉 


## Review notes
**Collection template**
_Vertical layout, ❌ Enable filtering, ☑️ Enable sorting_  **AND**
_Vertical layout, ☑️ Enable filtering, ☑️ Enable sorting_
- [ ] NO JS: I still see the duplicate Sorting UI on mobile

_Drawer layout, ☑️ Enable filtering, ☑️ Enable sorting_
- [ ] 🆕 NO JS: Pills wraps too soon on Collection, it was fixed yesterday though (They also wrap too soon on Search, but isn't check so I figure that one wasn't covered yet.)


**Search template**
_Vertical layout, ❌ Enable filtering, ☑️ Enable sorting_ **AND**
_Vertical layout, ☑️ Enable filtering, ☑️ Enable sorting_
- [ ] No JS: Still see the [duplicate of sorting visible ](https://screenshot.click/22-04-kiq7j-xmo1m.png)

**Comment by martinamarien** [comment]
Re those review notes Melissa. 
They should now be fixed in my PR 🤞🏻

**Review by ludoboludo — COMMENTED** [review]
Noted a few things but not sure if they need to be tackled and if they do if it would be within this PR. I'll leave it up to you: 
- [ ] spacing for the pills when vertical. I think the bottom spacing could match the spacing applied for the filters: [screenshot](https://screenshot.click/05-10-dyxwe-30s0x.png)
- [ ] the pills container on mobile seem like it could be sized differently instead of applying negative margin on it maybe 🤔 We're not ending up with a wider container as I would expect: [screenshot](https://screenshot.click/05-15-fff7s-2lhum.png)
- [ ] when filters are set to be vertical and sorting is disabled the total of products is gone: [screenshot](https://screenshot.click/05-19-esaqu-khfgh.png)
- [ ] I find the font size on desktop too big. It's matching the rest of the text while on mobile it drops to 10px. I kinda like the smaller font on pills for the hierarchy of the information provided. Plus the pills text color isn't getting any opacity so it comes out even more than any of the other info. [screenshot](https://screenshot.click/05-22-d16ae-3t8ns.png)
- [ ] I think it might be a bit odd that when setting filtering and sorting to drawer on desktop, it doesn't actually end up being a drawer when filtering is disabled: [screenshot](https://screenshot.click/05-30-kzta7-ot02l.png)
- [ ] some of the spacing is a bit different when set to horizontal or vertical. I had the filters disabled and when switching from one layout to another, you can see the different position/spacing of the sorting: [video](https://screenshot.click/05-29-gs62j-iioeu.mp4)
- [ ] one unlikely use case and specific to the admin, if you enable filtering, apply a filter then disable filtering the pills are staying there until you save or refresh your page: [screenshot](https://screenshot.click/05-36-03zmb-9v3jg.png) but I don't think it's likely to be a big issue. 
- [ ] In the no JS layout, I find the spacing could be better for the `filter` button when set to vertical: [screenshot](https://screenshot.click/05-49-y8cby-8wssa.png). I feel like having a link or button at the bottom of each filter/type would be actually the best in this scenario. [Something like this](https://screenshot.click/05-09-zqw9f-zt2ua.png)
- [ ] then a small little misalignment of the product count in no JS when filters are disabled and the layout is set to drawer : [screenshot](https://screenshot.click/05-52-zsoc5-ehs4t.png)

**Review by ludoboludo — COMMENTED** [review]
Just a few last comments. Some nitpicks. 

But ready to approve when some are answered :)

File: snippets/facets.liquid
Comment (martinamarien) [reply snippets/facets.liquid:386]: This one I cannot change. The menu drawer is used on mobile for all filter types. So in this case, it's checking for a desktop specific layout, which is independent from the mobile drawer.

Comment (ludoboludo) [reply snippets/facets.liquid:386]: Since it's part of the custom `<menu-drawer` element, I thought it would only apply to the drawer. So if this is coming up on desktop, then it means that the filter type is `drawer`.  

But this being a small nitpick, we can disregard 👍

**Review by ludoboludo — DISMISSED** [review]

**Pull request merged**
